src/utils/api.js

```diff
@@ -20,14 +39,49 @@ let newListing = {
   condition: "",
   location: "",
   tradeRequired: false,
-  user_id: ""
+  user_id: "",
 };
 
+let listingRef;
```

Comment:
this can be removed by passing the listing ref through the functions as an argument, see further comments for each stage